yt/frontends/enzo/data_structures.py

```diff
@@ -781,7 +781,7 @@ def _parse_enzo3_parameter_file(self, f):
 
     def _parse_enzo2_parameter_file(self, f):
         for line in (l.strip() for l in f):
-            if len(line) < 2: continue
+            if (len(line) < 2) or (line.lstrip()[0] == "#"): continue
```

Comment:
@brittonsmith correctly points out that there are some lines that do begin with `'#'` that we want to keep, e.g.:

```
#DataCGSConversionFactor[0] = 2.76112e-30
#DataCGSConversionFactor[1] = 1.32477e+06
#DataCGSConversionFactor[2] = 1.32477e+06
#DataCGSConversionFactor[3] = 1.32477e+06
#DataCGSConversionFactor[10] = 2.76112e-30
#DataCGSConversionFactor[11] = 2.76112e-30
#DataCGSConversionFactor[12] = 2.76112e-30
#DataCGSConversionFactor[13] = 2.76112e-30
#DataCGSConversionFactor[14] = 2.76112e-30
#DataCGSConversionFactor[15] = 2.76112e-30
#DataCGSConversionFactor[20] = 2.76112e-30
#DataCGSConversionFactor[21] = 2.76112e-30
#DataCGSConversionFactor[22] = 2.76112e-30
#DataCGSConversionFactor[23] = 2.76112e-30
#TimeUnits                 = 2.32946e+18
#TemperatureUnits          = 21261.6
```

So the real issue is a line that doesn't have an equal sign. So the real fix is to skip lines that don't have an equal sign.

Reply:
Apologies, I had just assumed that # at the start of the line meant a comment... The latest commit should have changed this to work now.

Reply:
No worries, I'm pretty sure I incorrectly said that on Slack without thinking very hard.

Even one line changes are tricky sometimes!